crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1373,15 +1373,31 @@ impl<'src> Lexer<'src> {
         }
 
         let mut current_position = self.current_range().start();
-        let reverse_chars = self.source[..current_position.to_usize()].chars().rev();
+        let mut reverse_chars = self.source[..current_position.to_usize()]
+            .chars()
+            .rev()
+            .peekable();
         let mut newline_position = None;
 
-        for ch in reverse_chars {
+        while let Some(ch) = reverse_chars.next() {
             if is_python_whitespace(ch) {
                 current_position -= ch.text_len();
             } else if matches!(ch, '\n' | '\r') {
                 current_position -= ch.text_len();
-                newline_position = Some(current_position);
+                if let Some(first_slash) = reverse_chars.next_if_eq(&'\\') {
+                    if let Some(second_slash) = reverse_chars.next_if_eq(&'\\') {
+                        // Line continuation character has been escaped: `\\\n`
+                        newline_position = Some(current_position);
+                        // Set the newline position before updating the current position.
+                        current_position -= first_slash.text_len() - second_slash.text_len();
+                    } else {
```

Comment:
I think it's still more complicate than this. What about `\\\` Here, we have an escaped backslash followed by a continuation :(

Reply:
I see. I guess we'd need to count the number of backslashes and make a decision based on whether it's odd or even.